### DISCUSSION TITLE
Integration with data acquisition software

### DISCUSSION THREAD

**Discussion by hbluhm** [qutech/qupulse#723] [General]
During the last meeting and today we discussed the migration from Matlab to Python and possible interfaces. One approach that deserves further study is to define a clean interface between data acquisition (similar or identical to current special-measure or qtlab) and the pulse-related control utilities.  Allowing a variety of data acquisition and instrument control solutions to be compatible could be attractive. In any case, a well defined interface with as few interconnections as possible seems desirable. 

As a start, we concluded that Simon would soon start exploring the used of the Matlab-Python interface to generate scan structs in Matlab from Python. This could provide a simple way to continue using our acquisition software together with the new framework.

The alternative is to migrate to QTlab. Here, first steps could be to implement a driver (e.g. for the averaging DLL) to see how well it suits our concept and perhaps a generic driver wrapper to simplify recycling.

In this context, I would like to bring this package to your attention: [labcontrol-software.com](http://labcontrol-software.com). It seems to support similar features as special measure, but is likely better documented, implemented less minimalistically and may be easier to use. A downside is that it seems to be on the way to being commercialized. One may debate whether a GUI is good or bad, but it does support Python scripting.

**Comment by pcerf** [comment]
Please find a board drawing from this discussion below. The interface Hendrik mentioned would be between MATLAB and Python, or between qtlab and Python (which would be nicer because it is Python to Python).
![Board](https://cloud.githubusercontent.com/assets/10511735/7489188/57d720de-f3d4-11e4-9eee-1997f29c7184.jpeg)
